docs/core/whats-new/dotnet-7.md

```diff
@@ -0,0 +1,181 @@
+
+## Regular expressions
+
+.NET's [regular expression](../../standard/base-types/regular-expressions.md) library has seen significant functional and performance improvements in .NET 7:
+
+- The new option <xref:System.Text.RegularExpressions.RegexOptions.NonBacktracking?displayProperty=nameWithType> enables matching using an approach that avoids backtracking and guarantees linear-time processing in the length of the input. The nonbacktracking engine can't be used in a right-to-left search and a has a few other restrictions, but is fast for all regular expressions and inputs.
+
+- Regular expression source generators are new. Source generators build an engine that's optimized for *your* pattern at compile time, providing throughput performance benefits. The source that's emitted is part of your project, so you can view and debug it. In addition, a new source-generator diagnostic `SYSLIB1045` alerts you to places you use <xref:System.Text.RegularExpressions.Regex> that could be converted to the source generator. For more information, see [.NET regular expression source generators](../../standard/base-types/regular-expression-source-generators.md).
+
+- For case-insensitive searches, .NET 7 includes large performance gains. The gains come because specifying <xref:System.Text.RegularExpressions.RegexOptions.IgnoreCase?displayProperty=nameWithType> no longer calls <xref:System.Char.ToLower%2A> on each character in the pattern and on each character in the input. Instead, all casing-related work is done when the <xref:System.Text.RegularExpressions.Regex> is constructed.
+
+- <xref:System.Text.RegularExpressions.Regex> now supports spans for some APIs. The following new methods have been added as part of this support:
+
+  - <xref:System.Text.RegularExpressions.Regex.EnumerateMatches%2A?displayProperty=nameWithType>
+  - <xref:System.Text.RegularExpressions.Regex.Count%2A?displayProperty=nameWithType>
+  - <xref:System.Text.RegularExpressions.Regex.IsMatch(System.ReadOnlySpan{System.Char})?displayProperty=nameWithType> (and a few other overloads)
+
+For more information about these and other improvements, see the [Regular expression improvements in .NET 7](https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/) blog post.
+
+## .NET libraries
+
+Many improvements have been made to .NET library APIs. Some are mentioned in other, dedicated sections of this article. Some others are summarized in the following table.
```

Comment:
```suggestion
Many improvements have been made to .NET library APIs. Some are mentioned in other, dedicated sections of this article. While others are summarized in the following table.
```

Reply:
That makes it a sentence fragment so I think I'll keep it as-is.